Sources/NIO/SocketChannel.swift

```diff
@@ -136,12 +136,19 @@ final class SocketChannel: BaseSocketChannel<Socket> {
 
                     assert(self.isActive)
                     pipeline.fireChannelRead0(NIOAny(buffer))
-                    if mayGrow && i < maxMessagesPerRead {
+                    result = .some
+
+                    if buffer.writableBytes > 0 {
+                        // If we not filled the whole buffer with read(...) we should stop reading and wait until we get notified again.
```

Comment:
grammar nit: 'If we did not fill the whole buffer' ...

Reply:
fixed